src/tools/imgtool/library.h

```diff
@@ -55,24 +55,27 @@ union filterinfo
 
 typedef void (*filter_getinfoproc)(uint32_t state, union filterinfo *info);
 
-struct imgtool_dirent
+namespace imgtool
 {
-	char filename[1024];
-	char attr[64];
-	uint64_t filesize;
-
-	time_t creation_time;
-	time_t lastmodified_time;
-	time_t lastaccess_time;
-
-	char softlink[1024];
-	char comment[256];
-
-	/* flags */
-	unsigned int eof : 1;
-	unsigned int corrupt : 1;
-	unsigned int directory : 1;
-	unsigned int hardlink : 1;
+	struct dirent
+	{
+		std::string filename;
+		std::string attr;
+		uint64_t filesize;
+
+		time_t creation_time;
+		time_t lastmodified_time;
+		time_t lastaccess_time;
+
+		std::string softlink;
+		std::string comment;
+
+		// flags
+		bool eof : 1;
+		bool corrupt : 1;
+		bool directory : 1;
+		bool hardlink : 1;
```

Comment:
Please don't use bitfields, they're trouble in the making.  They prevent taking references, etc.